test/e2e/specs/editor/various/multi-block-selection.spec.js

```diff
@@ -0,0 +1,1367 @@
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedBlocks )
+			.toMatchObject( [ { name: 'core/column' } ] );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedBlocks )
+			.toMatchObject( [
+				{ name: 'core/column' },
+				{ name: 'core/column' },
+			] );
+
+		await page.keyboard.press( 'Backspace' );
+
+		// Expect both columns to be deleted.
+		await expect
+			.poll( editor.getBlocks )
+			.toMatchObject( [ { name: 'core/columns', innerBlocks: [] } ] );
+	} );
+
+	test( 'should multi-select from within the list block', async ( {
+		editor,
+		pageUtils,
+		multiBlockSelectionUtils,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '1' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/list',
+			innerBlocks: [
+				{
+					name: 'core/list-item',
+					attributes: { content: '1' },
+				},
+			],
+		} );
+		// Focus on the list item.
+		await editor.canvas
+			.getByRole( 'document', { name: 'Block: List item' } )
+			.getByRole( 'textbox' )
+			.click();
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedBlocks )
+			.toContainEqual(
+				expect.objectContaining( { name: 'core/paragraph' } )
+			);
+	} );
+
+	test( 'should select all from empty selection', async ( {
+		page,
+		editor,
+		pageUtils,
+		multiBlockSelectionUtils,
+	} ) => {
+		for ( let i = 1; i <= 2; i += 1 ) {
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: { content: `${ i }` },
+			} );
+		}
+
+		// Clear the selected block.
+		await page.keyboard.press( 'Escape' );
+		await page.keyboard.press( 'Escape' );
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedBlocks )
+			.toEqual( [] );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+
+		await page.keyboard.press( 'Backspace' );
+
+		// Expect both paragraphs to be deleted.
+		await expect.poll( editor.getBlocks ).toEqual( [] );
+	} );
+
+	test( 'should select title if the cursor is on title', async ( {
+		page,
+		editor,
+		pageUtils,
+		multiBlockSelectionUtils,
+	} ) => {
+		for ( let i = 1; i <= 2; i += 1 ) {
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: { content: `${ i }` },
+			} );
+		}
+
+		await editor.canvas
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.type( 'Post title' );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedBlocks )
+			.toEqual( [] );
+		await expect
+			.poll( () =>
+				page.evaluate( () => window.getSelection().toString() )
+			)
+			.toBe( 'Post title' );
+	} );
+
+	test( 'should multi-select in the ListView component with shift + click', async ( {
```

Comment:
This is now a combination of two test cases because they share the same initial state:

- `should multi-select in the ListView component with shift + click`
- `should multi-select in the ListView component with shift + up and down keys`